docs/upgrade_guide.md

```diff
@@ -0,0 +1,30 @@
+# Upgrade Guide
+
+This guide describes how to upgrade Open Service Mesh (OSM).
+
+## Prerequisites
+- Kubernetes cluster with OSM installed
+- The [helm 3 CLI](https://helm.sh/docs/intro/install/)
```

Comment:
It might be worth explaining here why we're going through helm to do the upgrade instead of the `osm` CLI